ddtrace/contrib/internal/asgi/middleware.py

```diff
@@ -248,6 +249,9 @@ async def wrapped_send(message):
                 if span and message.get("type") == "http.response.start" and "status" in message:
                     cookies = _parse_response_cookies(response_headers)
                     status_code = message["status"]
+                    if self.integration_config._obfuscate_404_resource and status_code == 404:
```

Comment:
```suggestion
                    if status_code == 404 and self.integration_config._obfuscate_404_resource:
```

nit that can be ignored, the `status_code == 404` is _probably_ (untested) the faster operation to check first, since it doesn't involve multiple attribute look-ups, so we can use that as a first/faster gate